src/Compilers/Core/Portable/Operations/OperationCloner.cs

```diff
@@ -564,7 +581,9 @@ public override IOperation VisitMethodBodyOperation(IMethodBodyOperation operati
 
         public override IOperation VisitDiscardOperation(IDiscardOperation operation, object argument)
         {
-            return new DiscardOperation(operation.DiscardSymbol, ((Operation)operation).OwningSemanticModel, operation.Syntax, operation.Type, operation.ConstantValue, operation.IsImplicit);
+            return new DiscardOperation(
+                operation is IPatternOperation pat ? pat.InputType : operation.DiscardSymbol?.Type,
```

Comment:
>operation is IPatternOperation pat ? pat.InputType : operation.DiscardSymbol?.Type, [](start = 16, length = 83)

I didn't understand why cloning would require such logic.
Assuming we agreed that discard operation should contain an `InputType`, then cloning should just copy it.

Reply:
This is disentangled in #32018

---
In reply to: [244032505](https://github.com/dotnet/roslyn/pull/31967#discussion_r244032505) [](ancestors = 244032505)